protos/sources/vega/api/v1/core.proto

```diff
@@ -343,3 +348,50 @@ message LastBlockHeightResponse {
   // the network chain id from which the block comes from
   string chain_id = 8;
 }
+
+// A request to retrieve the spam statistics of a party for the given epoch
+message GetSpamStatisticsRequest {
+  string party_id = 1;
+}
+
+// The statistics for a given spam policy
+message SpamStatistic {
+  // The current transaction count received from the party during this epoch for this policy
+  string count_for_epoch = 1;
+  // The maximum number of transactions allowed for this policy in an epoch
+  string max_for_epoch = 2;
+  // If blocked the timestamp when the party will be unblocked as RFC3339Nano
+  optional string until = 4;
+}
+
+// The vote statistics for the voting spam policies
+// which are calculated as a ratio of the total votes
+// that have been rejected.
+message VoteSpamStatistic {
+  // The current vote count received from the party for the given policy during this epoch
+  string count_for_epoch = 1;
+  // If blocked the timestamp when the party will be unblocked as RFC3339Nano
+  optional string until = 2;
```

Comment:
until is not very descriptive for what it actually is referring to.

Reply:
I am all for longer and more descriptive variable names on the interface.